packages/connect-popup/src/log.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from 'react';
+import styled, { ThemeProvider } from 'styled-components';
+import { createRoot } from 'react-dom/client';
+
+import { Button, P, THEME, variables } from '@trezor/components';
+import { ErrorBoundary } from '@trezor/connect-ui/src/support/ErrorBoundary';
+import { GlobalStyle } from '@trezor/connect-ui/src/support/GlobalStyle';
+import { InfoPanel } from '@trezor/connect-ui/src/components/InfoPanel';
+import { View } from '@trezor/connect-ui/src/components/View';
+
+interface ReactWrapperProps {
+    children: React.ReactNode;
+}
+
+const MAX_ENTRIES = 1000;
+
+const ThemeWrapper = ({ children }: ReactWrapperProps) => (
+    <ThemeProvider theme={THEME.light}>{children}</ThemeProvider>
+);
+
+const Layout = styled.div`
+    display: flex;
+    flex: 1;
+    height: 100%;
+
+    @media (max-width: 639px) {
+        flex-direction: column;
+    }
+`;
+
+const StyledP = styled(P)`
+    margin: 0 20%;
+    font-size: 15px;
+    font-weight: ${variables.FONT_WEIGHT.LIGHT}
+    color: #757575
+`;
+
+const Wrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    justify-content: space-evenly;
+    align-items: center;
+    text-align: center;
+    min-height: 70vh;
+    margin-top: 10vh;
+    width: 100%;
+`;
+
+const DownloadButton = ({ array, filename }: { array: any[]; filename: string }) => {
+    const downloadArrayAsFile = () => {
+        const data = JSON.stringify(array, null, 2);
+        const blob = new Blob([data], { type: 'application/json' });
+
+        // Temporary anchor element.
+        const a = document.createElement('a');
+        a.href = URL.createObjectURL(blob);
+        a.download = filename;
+        document.body.appendChild(a);
+
+        // Programmatically trigger a click event on the anchor element.
+        a.click();
+
+        // Remove the anchor element from the document body.
+        document.body.removeChild(a);
+
+        URL.revokeObjectURL(a.href);
+    };
+
+    return (
+        <Button data-test="@log-container/download-button" onClick={downloadArrayAsFile}>
+            Download Logs
+        </Button>
+    );
+};
+
+const logInConsole = (logs: any[]) => {
+    logs.forEach(log => {
+        const { prefix, css, message } = log;
+        console.log(`%c${prefix}`, css, ...message);
+    });
+};
+
+const useLogWorker = (setLogs: React.Dispatch<React.SetStateAction<any[]>>) => {
+    const logWorker = new SharedWorker('./workers/shared-logger-worker.js');
+    useEffect(() => {
+        logWorker.port.onmessage = function (event) {
+            const { data } = event;
+            switch (data.type) {
+                case 'get-logs':
+                    logInConsole(data.payload);
+                    setLogs(data.payload);
```

Comment:
one important thing to think about: we want to be sure that logs are sorted by timestamp (this is the reason we send timestamp). 

1. we can easily sort logs before user downloads them. no problem here
2. when printing to console, the situation is a little bit more delicate. what is printed can't be unprinted a logs are being fired from all diffrent parts of connect and we do not have guarantee that they are in correct order. I am not sure how probable this is. post message is async thats for sure but maybe the order in which post message calls were scheduled is preserved? 🤔. 

How sorting could be solved: maybe some debouncing mechanism could work here. something like:
- when new log entry arrives, only save it to local state
- every 3 seconds execute a function that filters out all entries older than now - 3 seconds, sorts them and prints them one by one to console, and saves some marker "printed" to each record.

is it worth the effort or is it nonsense? 😹

Reply:
I updated it so when logs are downloaded the file contains logs ordered by timestamp https://github.com/trezor/trezor-suite/pull/8623/files#diff-97ce77aebcc482bb61f437dead565b77d4f163fda42c067a9aea21b84d842a09R130

Regarding the the debouncing mechanism that you propose sounds interesting but I am not sure about it since we can debounce it 3 seconds and then there will be a log that will take for some reason 3.1 seconds and it will not be organized. What I mean is that I do not think we can do it "bullet-proof".
Something to think about.